ledger/heavy/executor/backuper.go

```diff
@@ -0,0 +1,290 @@
+	}
+
+	metaInfoFile := filepath.Join(dirHolder.tmpDir, b.config.MetaInfoFile)
+	err = writeBackupInfoFile(md5sum, pulse, b.lastBackupedVersion, currentBT, metaInfoFile)
+	if err != nil {
+		return 0, errors.Wrap(err, "writeBackupInfoFile return error")
+	}
+
+	return currentBT, nil
+}
+
+type tmpDirHolder struct {
+	tmpDir  string
+	tmpFile *os.File
+}
+
+func (t *tmpDirHolder) release(ctx context.Context) {
+	err := t.tmpFile.Close()
+	if err != nil {
+		inslogger.FromContext(ctx).Fatal("can't close backup file: ", t.tmpFile, err)
+	}
+
+	err = os.RemoveAll(t.tmpDir)
+	if err != nil {
+		inslogger.FromContext(ctx).Fatal("can't remove backup file: ", t.tmpDir, err)
+	}
+}
+
+func (t *tmpDirHolder) reopenFile() error {
+	if err := t.tmpFile.Close(); err != nil {
+		return errors.Wrapf(err, "can't close file %s", t.tmpFile.Name())
+	}
+
+	reopenedFile, err := os.OpenFile(t.tmpFile.Name(), os.O_RDONLY, 0)
+	if err != nil {
+		return errors.Wrapf(err, "can't open file %s", t.tmpFile.Name())
+	}
+
+	t.tmpFile = reopenedFile
+	return nil
+}
+
+func (t *tmpDirHolder) create(where string, pulse insolar.PulseNumber) (func(context.Context), error) {
+	tmpDir, err := ioutil.TempDir(where, "tmp-bkp-"+pulse.String()+"-")
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't create tmp dir: %s", where)
+	}
+
+	file, err := os.OpenFile(tmpDir+"/incr.bkp", os.O_CREATE|os.O_WRONLY|os.O_EXCL, 0600)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't create tmp file. dir: %s", tmpDir)
+	}
+
+	t.tmpDir = tmpDir
+	t.tmpFile = file
+
+	return t.release, nil
+}
+
+func (b *BackupMakerDefault) doBackup(ctx context.Context, lastFinalizedPulse insolar.PulseNumber) (uint64, error) {
+
+	dirHolder := &tmpDirHolder{}
+	closer, err := dirHolder.create(b.config.TmpDirectory, lastFinalizedPulse)
+	if err != nil {
+		return 0, errors.Wrap(err, "can't create tmp dir")
+	}
+	defer closer(ctx)
+
+	currentBkpVersion, err := b.prepareBackup(ctx, dirHolder, lastFinalizedPulse)
+	if err != nil {
+		return 0, errors.Wrap(err, "prepareBackup returns error")
+	}
+
+	currentBkpDirName := fmt.Sprintf(b.config.DirNameTemplate, lastFinalizedPulse)
+	currentBkpDirPath := filepath.Join(b.config.TargetDirectory, currentBkpDirName)
+	err = move(ctx, dirHolder.tmpDir, currentBkpDirPath)
+	if err != nil {
+		return 0, errors.Wrap(err, "move returns error")
+	}
+
+	err = waitForBackup(ctx, filepath.Join(currentBkpDirPath, b.config.ConfirmFile), b.config.BackupWaitPeriod)
+	if err != nil {
+		return 0, errors.Wrapf(err, "waitForBackup returns error. pulse: %d", lastFinalizedPulse)
+	}
+
+	return currentBkpVersion, nil
+}
+
+func (b *BackupMakerDefault) Do(ctx context.Context, lastFinalizedPulse insolar.PulseNumber) error {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	if !b.config.Enabled {
+		inslogger.FromContext(ctx).Info("Trying to do backup, but it's disabled. Do nothing")
+		return ErrBackupDisabled
+	}
+
+	if lastFinalizedPulse <= b.lastBackupedPulse {
```

Comment:
use !Equal here

Reply:
It's equal or less. Not just "not equal"